src/mol-model/structure/query/queries/atom-set.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2019 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Koya Sakuma
+ * Adapted from MolQL implemtation of atom-set.ts
+ *
+ * Copyright (c) 2017 MolQL contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author David Sehnal <david.sehnal@gmail.com>
+ */
+
+import { StructureQuery } from '../query';
+import { StructureSelection } from '../selection';
+import { getCurrentStructureProperties } from './filters';
+import { QueryContext, QueryFn } from '../context';
+
+
+export function atomCount(ctx: QueryContext) {
+    return (ctx: QueryContext) => {
+        const all = StructureSelection.Singletons(ctx.currentStructure, ctx.currentStructure);
+        const x: number = StructureSelection.structureCount(all);
+        return x;
+    };
+}
```

Comment:
You are adding an unnecessary layer of "context" here:

```ts
export function atomCount(ctx: QueryContext) {
  return ctx.currentStructure.elementCount;
}
```

and later

```ts
D(MolScript.structureQuery.atomSet.atomCount,
        function structureQuery_atomset_atomCount(ctx, xs) {
	    return Queries.atomset.atomCount(ctx);
        }),
```

Reply:
Thank you @dsehnal , I've adopted your suggestions and checked that they works fine.